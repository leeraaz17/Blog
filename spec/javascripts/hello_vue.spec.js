import { shallowMount, mount } from "@vue/test-utils";
import App from "@/app/javascript/app.vue";
import { createRenderer } from "vue-server-renderer";

describe("app", () => {
  it("success matches snapshot", () => {
    const renderer = createRenderer();
    const wrapper = shallowMount(App);
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});

describe('App', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(App)
    expect(wrapper.isVueInstance()).toBeTruthy()
    // expect(wrapper.attributes().message).toBe("Hello Vue!")
    // const hellos = JSON.parse(JSON.stringify(wrapper))
    // console.log(wrapper.html())
    // console.log(hellos)
  })

  it("should display props.ms when passed", () => {
    // const msg = "This is something new";
    // const wrapper = shallowMount(App, {
    //   propsData: { msg }
    // });
    // expect(wrapper.text()).toMatch(msg);
  })
})
