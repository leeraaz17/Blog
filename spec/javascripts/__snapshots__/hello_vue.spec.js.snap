// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app success matches snapshot 1`] = `"<div id=\\"app\\" data-server-rendered=\\"true\\"><p>Hello Vue!</p></div>"`;
